geode-protobuf/src/test/java/org/apache/geode/protocol/RoundTripCacheConnectionJUnitTest.java

```diff
@@ -291,56 +298,108 @@ public void testConnectionCountIsProperlyDecremented() throws Exception {
   }
 
   @Test
-  public void testNewProtocolRespectsMaxConnectionLimit() throws IOException, InterruptedException {
-    cache.getDistributedSystem().disconnect();
+  public void testNewProtocolRespectsMaxConnectionLimit_notSelector() throws Exception {
+    testNewProtocolRespectsMaxConnectionLimit(0, false);
+  }
+
+  @Test
+  public void testNewProtocolRespectsMaxConnectionLimit_isSelector() throws Exception {
+    testNewProtocolRespectsMaxConnectionLimit(4, true);
+  }
+
+  private void testNewProtocolRespectsMaxConnectionLimit(int threads, boolean isSelector)
+      throws Exception {
+    final int connections = 16;
+
+    cache.close();
```

Comment:
why do we close the cache, when the only thing we want to achieve is that we want to start a cacheserver with a specific connection limit. Would the following code not be a little simpler?

`   cache.getCacheServers().get(0).stop();

    CacheServer cacheServer = cache.addCacheServer();
    final int cacheServerPort = AvailablePortHelper.getRandomAvailableTCPPort();
    cacheServer.setPort(cacheServerPort);
    cacheServer.setMaxConnections(16);
    cacheServer.setMaxThreads(16);
    cacheServer.start();`